File: 1D Array/C/program_8.c
#include <stdio.h>

void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

void waveSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i += 2) {
        if (i > 0 && arr[i] < arr[i - 1]) {
            swap(&arr[i], &arr[i - 1]);
        }
        if (i < n - 1 && arr[i] < arr[i + 1]) {
            swap(&arr[i], &arr[i + 1]);
        }
    }
}

int main() {
    int arr[] = {10, 5, 6, 3, 2, 20, 100, 80};
    int n = sizeof(arr) / sizeof(arr[0]);
    
    waveSort(arr, n);
    
    printf("Wave Form Sorted Array: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
    
    return 0;
}

File: 1D Array/java/program8.java
public class program8 {
    public static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    public static void waveSort(int[] arr, int n) {
        for (int i = 0; i < n - 1; i += 2) {
            if (i > 0 && arr[i] < arr[i - 1]) {
                swap(arr, i, i - 1);
            }
            if (i < n - 1 && arr[i] < arr[i + 1]) {
                swap(arr, i, i + 1);
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {10, 5, 6, 3, 2, 20, 100, 80};
        int n = arr.length;
        
        waveSort(arr, n);
        
        System.out.print("Wave Form Sorted Array: ");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
}